frontend/src/views/WheresMyGene/components/Filters/components/Compare/connect.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  DefaultDropdownMenuOption,
+  InputDropdownProps as IInputDropdownProps,
+} from "@czi-sds/components";
+import { useCallback, useContext, useMemo } from "react";
+import { track } from "src/common/analytics";
+import { EVENTS } from "src/common/analytics/events";
+import { COMPARE_OPTIONS } from "src/views/WheresMyGene/common/constants";
+import {
+  DispatchContext,
+  State,
+  StateContext,
+} from "src/views/WheresMyGene/common/store";
+import { selectCompare } from "src/views/WheresMyGene/common/store/actions";
+import { Props } from "./types";
+
+export const useConnect = ({ areFiltersDisabled }: Props) => {
+  const dispatch = useContext(DispatchContext);
+  const { compare } = useContext(StateContext);
+
+  const DEFAULT_INPUT_DROPDOWN_PROPS: Partial<IInputDropdownProps> = useMemo(
+    () => ({
+      sdsStyle: "square",
+    }),
+    []
+  );
+
+  const InputDropdownProps = useMemo(
+    () => ({
+      ...DEFAULT_INPUT_DROPDOWN_PROPS,
+      disabled: areFiltersDisabled,
+    }),
+    [areFiltersDisabled, DEFAULT_INPUT_DROPDOWN_PROPS]
+  );
+
+  const optionLabel = useMemo(() => {
+    return COMPARE_OPTIONS.find((option) => option.id === compare);
+  }, [compare]);
+
+  const handleChange = useCallback(
+    (value: DefaultDropdownMenuOption | null) => {
+      let groupByEventTriggeredAtLeastOnce = false;
```

Comment:
I think this was there to prevent firing an analytics event when the page first loads per `// (alec) This is a hack to prevent the group by event from firing when the page loads` - if we still need this, we might need to hoist this variable to the top level, otherwise this variable will be created again with each callback invocation!

Or if after your refactoring this is no longer an issue, then we can just remove this?

Reply:
removed 👯